hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -272,6 +272,25 @@ object PruneDeadFields {
             // don't memoize right if we are going to elide it during rebuild
             memoizeTableIR(left, requestedType, memo)
         }
+      case TableIntervalJoin(left, right, root) =>
+        val fieldDep = requestedType.rowType.fieldOption(root).map(_.typ.asInstanceOf[TStruct])
```

Comment:
this is just copied from TableLeftJoinRightDistinct right?

Reply:
Yes.  Lazy, I know, although it isn't totally clear how to share the code since pattern matching can't share bindings between alternatives.  Suggestions?